api/v1alpha1/nodehealthcheck_types.go

```diff
@@ -34,7 +34,7 @@ type NodeHealthCheckSpec struct {
 	// logical OR, i.e. if any of the conditions is met, the node is unhealthy.
 	//
 	// +optional
-	// +kubebuilder:default:={{type:Ready,status:False,duration:"300s"}}
+	// +kubebuilder:default:={{type:Ready,status:False,duration:"300s"},{type:Ready,status:Unknown,duration:"300s"}}
```

Comment:
does this default mean that if the user hasn't specified these conditions they will be applied automatically?
does it mean a user can't really ignore/not-use this conditions?

Reply:
it means that if not specified those are the defaults. You can have a look a the test

Reply:
once you have those defaults, you can't configure NHC CR without those.
so I'm only interested in NHC CR with DiskPressure, but without doing anything to the `Ready` condition, it won't be possible

Reply:
its only if you omitted unhealthyConditions from the yaml.

```shell
$ cat << EOF | kubectl create -f -
apiVersion: remediation.medik8s.io/v1alpha1
kind: NodeHealthCheck
metadata:
  namespace: default
  name: nodehealthcheck-sample
spec:
  unhealthyConditions:
    - type: DiskPressure
      status: "True"
      duration: 300s
  remediationTemplate:
    kind: PoisonPillRemediationTemplate
    apiVersion: poison-pill.medik8s.io/v1alpha1
    name: ppill-template
    namespace: default
EOF
nodehealthcheck.remediation.medik8s.io/nodehealthcheck-sample created
[rgolan@rgolan ~/src/node-healthcheck-operator (update-defaults %<>)] 
$ k get nhc
NAME                     AGE
nodehealthcheck-sample   5s

$ k get nhc nodehealthcheck-sample -o jsonpath={.spec.unhealthyConditions} | jq
[
  {
    "duration": "300s",
    "status": "True",
    "type": "DiskPressure"
  }
]

```

Reply:
Ah very cool